### PR TITLE
Add drone-k8s-helm Plugin

### PR DESCRIPTION
**Pull request by f-a-a**
WIP

### CODE REVIEW

File: content/127labs/drone-k8s-helm/index.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    clean_before_release: true
+    values:
+      dokuwikiEmail: $${DOKUWIKI_EMAIL}
+      dokuwikiPassword: $${DOKUWIKI_PASSWORD}
+    secrets: [dokuwiki_email, dokuwiki_password]
+```
+
+# Secret Reference
+
+Values are capable of accessing secrets by interpolating them with the `$${SECRET}` syntax. The plugin will
```
Comment (bradrydzewski) [inline content/127labs/drone-k8s-helm/index.md]: Could you explain this a bit more? This seems to work a bit differently than most plugins and might therefore be a bit confusing to users. Perhaps if I can understand this a bit better, I can suggest a way to better document or even perhaps better implement such a feature (if you are open to it).

File: content/127labs/drone-k8s-helm/index.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+---
+date: 2017-06-10T00:00:00+00:00
+title: Kubernetes Helm
+author: 127labs
+tags: [ kubernetes, helm, docker ]
+repo: 127labs/drone-k8s-helm
+logo: kubernetes_helm.svg
```
Comment (bradrydzewski) [inline content/127labs/drone-k8s-helm/index.md]: can you set to `helm.svg` since we already have a helm logo in the repository? thanks!

Comment (f-a-a) [inline content/127labs/drone-k8s-helm/index.md]: sorry I overlooked this, I've fixed it.

File: content/127labs/drone-k8s-helm/index.md (hunk 1 above)
Comment (f-a-a) [inline content/127labs/drone-k8s-helm/index.md]: The motivation here was to allow `PLUGIN_VALUES` env var to access secrets. We found out that by setting the `values` key (in 0.7) this way:
```yaml
values:
  dokuwikiEmail: $${DOKUWIKI_EMAIL}
  dokuwikiPassword: $${DOKUWIKI_PASSWORD}
```
is being translated into:
```
PLUGIN_VALUES={"dokuwikiEmail":"${DOKUWIKI_EMAIL}","dokuwikiPassword":"${DOKUWIKI_PASSWORD}"}
```
and if we have secrets:
```
DOKUWIKI_EMAIL=john@doe.com
DOKUWIKI_PASSWORD=password
```
being set for example. They will be available as env variable where `PLUGIN_VALUES` has access to.

I hope the explanation was clear and yes please, would really love to have your input in this.

Comment (bradrydzewski) [inline content/127labs/drone-k8s-helm/index.md]: thanks for the reply!

One of the things that jumped out is the use of `${VARIABLE}`. In older versions of drone we injected secrets into the yaml using this exact syntax. We also use this syntax to inject build metadata into the yaml (such as branch, sha, etc). So there is a bit of history with this syntaxthat could confuse people.

One option could be separating hard-coded variables and variables sourced from the environment. The latter would map environment variables to use-defined names.

```diff
values:
  dokuwikiEmail: 'hardcodedEmail'
  dokuwikiPassword: 'hardcodedPassword'
+values_from_env:
+  dokuwikiEmail: DOKUWIKI_EMAIL
+  dokuwikiPassword: DOKUWIKI_PASSWORD
```

So for the record, nothing wrong with the implementation. More just the history of that specific syntax that tripped me up initially :)

Comment (f-a-a) [inline content/127labs/drone-k8s-helm/index.md]: What do you think if we use a different syntax instead? (i.e. `{{DOKUWIKI_EMAIL}}`) would it be clearer?

Or do you think it is okay to maintain the syntax but describe in more detail in the documentation?

Comment (bradrydzewski) [inline content/127labs/drone-k8s-helm/index.md]: also I noticed this in the github readme:

```
    secrets:
      - dokuwiki_email
      - dokuwiki_password
      - plugin_master
      - plugin_token
```

`PLUGIN_` is a convention used to namespace variables that are defined in the yaml. In the future we may not let users set variables with `PLUGIN_` so you may not be able to rely on using this value for user-defined secrets.

If you look at the docker plugin (for example) you will notice we source the environment variable name from `PLUGIN_` and from a more user-friendly environment variable which the end-user can use in conjunction with secrets.

```
		cli.StringFlag{
			Name:   "docker.username",
			Usage:  "docker username",
			EnvVar: "PLUGIN_USERNAME,DOCKER_USERNAME",
		},
		cli.StringFlag{
			Name:   "docker.password",
			Usage:  "docker password",
			EnvVar: "PLUGIN_PASSWORD,DOCKER_PASSWORD",
		},
```

PS sorry if this is all a bit confusing. I haven't had time to put together a best practices document for plugin authors

**Pull request closed without merging**